1. Array/weird_sorting/Weird_Sorting_Arnab.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * WeirdSortingArnab.java
+ * Given an array of integers containing 7,14 and 21.Sort them in a non-decreasing manner with O(n) time and space 
+   complexity
+ * Description:-
+ * Here we keep track of 2 indices-left(for 7) and right(for 21) using 2 pointer approach, and then just run a loop from'
+ * 0 to n-1 and swap when 7 and 21 are encountered and just increment i when 14 is encountered.
+ * Finally we would achieve the formation which we want as 14 is not altered it would come in the middle. 
+ * Time Complexity-O(n) Extra Space Complexity-O(1)
+ * @author [codebook-2000] (https://github.com/codebook-2000)
+ */
+ 
+
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+
+class Weird_Sorting_Arnab
```

Comment:
Please change the name of the class and file as per Java convention

Reply:
<h2>I have updated it, Kindly check it</h2>